app/common/data/interfaces/exceptions.py

```diff
@@ -9,7 +9,10 @@ class DuplicateValueError(Exception):
     field_name: str
     new_value: str
 
-    constraint_name_map: dict[str, str] = {"uq_grant_name": "name"}
+    # I think this should probably just wrap the IntegrityError and rely on its message for diagnosing
+    # presumably most things calling this will know what the integrity failure is and if they don't they could
+    # inspect the constraint name themselves (in the rare instance there might be multiple)
```

Comment:
I think this is a useful comment to put on a PR and then work through, but not useful to have in the code as an unresolved thought? Unless you're saying it's a decision and we should do it rather than just a musing.

Reply:
Yeah agree this wouldn't be to stick here permanently but was just pushing this as WIP -- I think either the PR makes the actual change or its just removed entirely and we stick with it (so I put it somewhere I wouldn't forget - happy to avoid that if they start to make it in or we'd prefer not to).

I think I wasn't expecting to update an error signature while adding a database constraint but tbh this mapping has to happen somewhere and I don't have that strong an opinion on it